analyzer/evmnfts/httpmisc/http_misc.go

```diff
@@ -0,0 +1,49 @@
+package httpmisc
+
+import (
+	"context"
+	"fmt"
+	"net/http"
+	"time"
+)
+
+const ClientTimeout = 30 * time.Second
+
+func GetWithContextWithClient(ctx context.Context, client *http.Client, url string) (*http.Response, error) {
+	req, err := http.NewRequestWithContext(ctx, "GET", url, nil)
+	if err != nil {
+		return nil, err
+	}
+	return client.Do(req)
+}
+
+type NotOKError struct {
+	// Note: .error is the implementation of .Error, .Unwrap etc. It is not
+	// in the Unwrap chain. Use something like
+	// `NotOKError{fmt.Errorf("...: %w", err)}` to set up an
+	// instance with `err` in the Unwrap chain.
+	error
+}
+
+func (err NotOKError) Is(target error) bool {
+	if _, ok := target.(NotOKError); ok {
+		return true
+	}
+	return false
+}
+
+func ResponseOK(resp *http.Response) error {
+	if resp.StatusCode >= 500 {
```

Comment:
I don't understand how this is different from `StatusCode != 200`; the handler seems identical, except for the subtle difference in producing the `NotOKError` wrapper or not.

Reply:
answering here https://github.com/oasisprotocol/nexus/pull/506#discussion_r1339330308

Reply:
closing this, let's discuss in the other conversation :skull: